pkg/resources/nifi/service.go

```diff
@@ -98,12 +99,17 @@ func (r *Reconciler) generateServicePortForExternalListeners(eService v1.Externa
 	for _, port := range eService.Spec.PortConfigs {
 		for _, iListener := range r.NifiCluster.Spec.ListenersConfig.InternalListeners {
 			if port.InternalListenerName == iListener.Name {
-				usedPorts = append(usedPorts, corev1.ServicePort{
+				newPort := corev1.ServicePort{
 					Name:       strings.ReplaceAll(iListener.Name, "_", ""),
 					Port:       port.Port,
 					TargetPort: intstr.FromInt(int(iListener.ContainerPort)),
 					Protocol:   corev1.ProtocolTCP,
-				})
+				}
+
+				if eService.Spec.Type == corev1.ServiceTypeNodePort && port.NodePort != nil {
```

Comment:
If the service type is node port and the user doesn't set a `PortConfig.NodePort`, then should we fail reconciliation with an error indicating this?

Does it make sense to configure a node port service, but then no node ports?

Reply:
This is to preserve the current behavior. If you set the service type to `NodePort`, the `nodePort` will be chosen randomly by the Kubernetes control plane (see https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport). I've just added the ability to specify this, but you can leave it out and have a random port as it is now.

Reply:
Cool - sounds good!